lib/Pipes/TaggedFunctionKind.cpp

```diff
@@ -49,18 +55,223 @@ TaggedFunctionKind::symbolToTarget(const llvm::Function &Symbol) const {
+        // Invalidate all targets
+        for (const model::Function &Function : Model->Functions)
+          Targets.insert(Function.Entry.toString());
+      } else if (Change.Old) {
+        // Invalidate the function and its callers
+        auto Function = std::get<model::Function>(*Change.Old);
+        Targets.insert(Function.Entry.toString());
+        insertCallers(CrossRelations, Function.Entry, Targets);
+      }
+    }
+
+    // A new prototype has been added or removed?
+    if (pathAsString<Binary>(Change.Path) == "/Types") {
+      if (Change.New) {
+        // Invalidate all targets
+        for (const model::Function &Function : Model->Functions)
+          Targets.insert(Function.Entry.toString());
+      } else if (Change.Old) {
+        auto Prototype = std::get<model::TypePath>(*Change.Old);
+
+        for (const model::Function &Function : Model->Functions) {
+          if (Prototype == Function.Prototype) {
+            // Invalidate the function and its callers
+            Targets.insert(Function.Entry.toString());
+            insertCallers(CrossRelations, Function.Entry, Targets);
+          }
+        }
+
+        for (const model::DynamicFunction &Function : IDF) {
+          if (Prototype == Function.Prototype) {
+            // Invalidate callers of the model::DynamicFunction
+            insertCallers(CrossRelations, Function.OriginalName, Targets);
+          }
+        }
+      }
+    }
+
+    // Attributes have changes?
+    auto MaybeChangedAttributes = AttrMatcher.match<MetaAddress>(Change.Path);
+    if (MaybeChangedAttributes) {
+      auto Entry = std::get<MetaAddress>(*MaybeChangedAttributes);
+      using FunctionAttribute = model::FunctionAttribute::Values;
+      bool NewAttribute = std::get<FunctionAttribute>(*Change.New);
+      if (NewAttribute == FunctionAttribute::NoReturn) {
+        // Invalidate callers
+        insertCallers(CrossRelations, Entry, Targets);
+      }
+    }
+
+    // A new dynamic function has been added or removed?
+    if (pathAsString<Binary>(Change.Path) == "/ImportedDynamicFunctions") {
+      if (Change.New) {
+        // Invalidate all targets
+        for (const model::Function &Function : Model->Functions)
+          Targets.insert(Function.Entry.toString());
+      } else if (Change.Old) {
+        // Invalidate callers of the model::DynamicFunction
+        for (const model::DynamicFunction &Function : IDF)
+          insertCallers(CrossRelations, Function.OriginalName, Targets);
+      }
+    }
+  }
+
+  // Because matching e.g., FinalStackOffset, for both RawFunctionType and
+  // CABIFunctionType is non-trivial, we will not leverage the diff for checking
+  // the change related to FSO; instead, we pass to `Layout` by employing the
+  // old model, and the new one (the one with the change).
+  for (const model::Function &NewFunction : Model->Functions) {
+    MetaAddress Entry = NewFunction.Entry;
+    const auto &OldFunction = OldModel->Functions.at(Entry);
```

Comment:
The function might have been removed. You can't use `.at`.

Reply:
Wait sorry, I still have the function in the old model, no? Otherwise, what's the purpose of checking `FSO` & stuff if I do not have the function?